### PR TITLE
Use initialValue in withOnyx and other optimizations for ReportScreen

### PR DESCRIPTION
**Pull request by ospfranco**
### Details

This PR makes further improvements to the ReportScreen. It is based on the changes of the previous [PR](https://github.com/Expensify/App/pull/26173). 

A summary of the changes:
- A lot of re-renders were caused by react-native-onyx delaying the rendering because it had to fetch data from the DB. This was solved by adding a `initialValue` to the withOnyx HOC. This allows components to mount ASAP. This had to be manually added to the components that were causing a re-render.
- Large components with a lot of onyx connections would also block rendering as data was being fetched/hydrated to the component. This was a particular case of ReportScreen. A solution was implemented where we tell onyx to delay updates until the component has been mounted. This allows React to perform a render instead of blocking completely.
- ReportScreenWrapper was using an internal hook to update route params. This was moved into its own component to avoid a large tree-rerender
- `HeaderView.js` had unused onyx connections and was returning a new object every time. Causing a re-render
- Some other minor optimizations like skipping some hooks on first render.

This PR is also dependant on changes we made to `react-native-onyx` in this [PR](https://github.com/Expensify/react-native-onyx/pull/329) those need to be merged and published first.

### Fixed Issues

$ https://github.com/Expensify/App/issues/26087

### Tests

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by janicduplessis** [comment]
I did another round on profiling on this branch and found some more improvements

### Prevent SidebarLinks from re-rendering on opening report

`currentReportID` is passed as a prop down to `SidebarLinks` and causes a re-render of the whole list when opening a new report. Turns out `SidebarLinks` only uses `currentReportID` in the `showReportPage` function which is invoked when clicking a report and never in the render path so we can use `Navigation.getTopmostReportId` to get it without having to pass it as prop and causing a re-render. We also have onyx connected to fetch the current report which is never used so we can remove that too. After that `SidebarLinks` won't re-render on opening a report since it is a pure component and props do not change even if parent `SidebarLinksData` re-renders.

Before:

image

After:

image

The perf impact is pretty minimal, but it is also a good change to cleanup `SidebarLinks`. Ideally `SidebarLinksData` would not re-render at all but `getOrderedReportIDs` is dependent on the `currentReportID`.

Diff: https://github.com/janicduplessis/Expensify-App/commit/b47663e2fd63bb056b63b617a7893869bbacf488

### Reduce SidebarLinksData render time 

I was able to get this render down more significantly by changing the memoization to separate `getOrderedReportIDs` from `currentReportID`, but this will cause slight behaviour change since if for some reason the opened report is not in the reports list, it will be added at the start instead of being added where it should be according to the sorting logic. I feel like the case where the opened report is not visible in the report list should be very rare, but I am not familiar enough with the product to know for sure. Another option to preserve 100% behaviour at the cost of performance would be to compute the reports list a 2nd time if the current report is not visible. This will result in 2x worst perf for the edge case of the current report not being visible, but great improvement in the common case. I think someone with more product knowledge could make the call on this one.

image

Diff: https://github.com/janicduplessis/Expensify-App/commit/2c14b627513ae88b92c50598961d9c04cc5a8f0a

Will update this as I find more things

**Comment by ospfranco** [comment]
I applied the first performance update suggested by @janicduplessis, tested it and everything is working fine.

I will wait for the green light from someone in expensify to approve the change in the second suggestion.

**Comment by janicduplessis** [comment]
@ospfranco I will open a new PR for the 2nd perf improvement so it can be reviewed separately. We concluded on slack that we can preserve the exact same behviour at the cost of generating the list a 2nd time for the edge case.

**Comment by janicduplessis** [comment]
https://github.com/Expensify/App/pull/27364

**Comment by mountiny** [comment]
Could you sync this up with https://github.com/Expensify/App/pull/26173 and then close that PR?

**Comment by ospfranco** [comment]
@mountiny Already done

**Comment by mountiny** [comment]
Thanks!

**Comment by ospfranco** [comment]
The Onyx PR has been merged, which means this is clear now. I need to update onyx in the main app though, how do you want to proceed @mountiny ? create a new PR for bumping onyx or should I update the onyx dependency on this PR?

**Comment by mountiny** [comment]
@ospfranco so its a bit tricky because we already have the previous onyx change for batching https://github.com/Expensify/App/pull/27230 and basically if we merge and test this PR we are combining all the changes so I think we need to be patient and first thorughly test the mentioned PR with batched updated, fix issues there and then focus on QAing this one and getting it out

cc @tgolen @marcaaron if you interested in review here.

**Comment by tgolen** [comment]
Yeah, let's hold off on updating App the batched update PR testing has passed. If this is critical to get out (and I don't think it is), we could revert the batch update change and then push this change out.

**Comment by mountiny** [comment]
I think they build on top of each other so makes sense to just take it one by one

**Comment by mountiny** [comment]
Triggered a new build on this one. @fedirjh would you be able to do testing on this PR today? thanks!

**Comment by mountiny** [comment]
@ospfranco unit tests are failing

**Comment by ospfranco** [comment]
Had one small mistake when I merged main. It's fixed now.

**Comment by mountiny** [comment]
Triggered a new build

**Comment by ospfranco** [comment]
There currently seems to be an issue. When the page first loads the skeleton view never gets unmounted, this was due to a cyclic condition I fixed [here](https://github.com/Expensify/App/pull/26772/commits/492f847995c7d74b8b2d6ffc32d25943679ada24). I can no longer reproduce the issue on a local build, however on the testing environment (https://26772.pr-testing.expensify.com/) is still happenning to me for some reason. How can I check which commit is deployed in the test env?


CleanShot 2023-09-21 at 11 17 56

**Comment by mountiny** [comment]
@ospfranco Navigating to a thread you have not visited yet shows Hmm its not there page

**Comment by ospfranco** [comment]
Taking a look

**Comment by ospfranco** [comment]
Latest commit seems to have fixed it @mountiny

**Comment by mountiny** [comment]
Testing will be done on firday and over the weekend and they will post results here

**Comment by kbecciv** [comment]
Issue is reproduced from previous special build https://github.com/Expensify/App/pull/27230#issuecomment-1723296484

**Comment by izarutskaya** [comment]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
### Web - Request Money - Next button is not grayed out when Request Money without an amount entered

## Action Performed:
1. Open https://26772.pr-testing.expensify.com/
2. Click on the green plus button (FAB)
3. Select Request Money

## Expected Result:
"Next" button should be grayed out since there is no amount entered

## Actual Result:
Next button is not grayed out with no amount entered

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** v1.3.72-9 PR:26772

**Reproducible in staging?:** No

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/ce025351-cf9f-49b3-8d28-d0e3b208ac1c

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
### Web – Attachment - Unable to add PDF file

## Action Performed:
1. Open https://26772.pr-testing.expensify.com/
2. Log in with any Gmail account.
3. Open 1:1 conversation, click on the + button in the compose box.
4. Click on add attachment.
5. Upload a PDF..

## Expected Result:
Preview of the PDF is displayed before uploading to the conversation.

## Actual Result:
Black page opens with the message “Failed to load PDF file”.

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** v1.3.72-9 PR:26772

**Reproducible in staging?:** No

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/8a5eb681-2841-4d43-aa6d-56a260785530

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
### Web - Task - Task details are not cleared immediately after deleting the task

## Action Performed:
1. Go to [https://26772.pr-testing.expensify.com/.](https://26772.pr-testing.expensify.com/)
2. Go to + > Assign task.
3. Create a task.
4. In the task report, click on the menu > Cancel.

## Expected Result:
The task details are cleared immediately.

## Actual Result:
The task details are not cleared immediately unless user navigates to the chat and then back to the task report.

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** v1.3.72-9 PR:26772

**Reproducible in staging?:** No

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/6d8ddfe4-d365-4c46-9118-31c5cff5e65f

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
### Web – Deeplinks – Unable to navigate to IOU via link.

## Action Performed:
1. Go to [https://26772.pr-testing.expensify.com/.](https://26772.pr-testing.expensify.com/)
2. Log in with any user.
3. As account A, create an IOU with account B.
4. Navigate to the IOU details page and copy the URL.
5. Log out of account A.
6. Log in to account B.
7. Paste and navigate to the same IOU link you can in the clipboard.

## Expected Result:
Redirected to the IOU details while logged in as Account B.

## Actual Result:
Redirected to the error page.

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** v1.3.72-9 PR:26772

**Reproducible in staging?:** No

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/59d63866-3ed8-4ccb-9e7e-0ab727525fe7

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
### Web – Deeplinks – Unable to navigating to Public Room as anon user

## Action Performed:
1. Go to [https://26772.pr-testing.expensify.com/.](https://26772.pr-testing.expensify.com/)
2. Log out.
3. Navigate to the link https://26772.pr-testing.expensify.com/r/5408450846930023.


## Expected Result:
Verify you can see the announce room as an anonymous user

## Actual Result:
Redirect to Log in page.

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** v1.3.72-9 PR:26772

**Reproducible in staging?:** No

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/b5727adb-4632-4c12-8cc9-15c31da30e0d

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
### IOS - Unable login with Apple error displayed

## Action Performed:
1. Launch App
2. Tap sign in with Apple
3. Tap Continue on popup

## Expected Result:
Able to Sign In with Apple

## Actual Result:
Unable to Sign In with Apple: error "Cannot get account details, please try again or contact [concierge@expensify.com"](mailto:concierge@expensify.com%22)

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** v1.3.72-9 PR:26772

**Reproducible in staging?:** No

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/cd57856a-91b3-4b6a-a6bf-093c2fe995d0

Bug6211936_apple

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

File: src/pages/home/report/ReportFooter.js
Comment (ospfranco) [reply src/pages/home/report/ReportFooter.js:105]: In this case we believed not, otherwise the composer would be shown on every single page that gets opened. Since this was blocked to render since it was waiting for onyx, the default value would not matter, but now it does, wdyt?

File: src/pages/home/ReportScreen.js
Comment (ospfranco) [reply src/pages/home/ReportScreen.js:327]: Why would it run twice? from our discussions with the mounting of components, new chats will be pushed into the stack right?

In any case, once the component has been cleared to receive onyx updates, calling this again is a noop. So it's fine.

**Comment by mountiny** [comment]
Super excited for this PR, lets resolve the issues found 🚀 @fedirjh are you able to help @ospfranco with them?

**Comment by ospfranco** [comment]
Hey, I just went through the issues. Here is my findings. @fedirjh maybe you can give me a hand trying to replicate the issues I could not reproduce.

- **Web - Request Money - Next button is not grayed out when Request Money without an amount entered**
    - This is also happening on main/staging, tested it. Has nothing to do with my changes.
- **Web – Attachment - Unable to add PDF file**
    - Not able to reproduce this locally. Only on the ad-hoc build it seems to be failing. @mountany do you know better what could be going on.
- **Web - Task - Task details are not cleared immediately after deleting the task**
    - Fixed
- **Web – Deeplinks – Unable to navigate to IOU via link.**
    - Could not reproduce locally. Can somebody else also test this?
- **Web – Deeplinks – Unable to navigating to Public Room as anon user**
    - Not reproducible locally. Can somebody else also test this?
        
        
        
- **IOS - Unable login with Apple error displayed**
    - This has nothing to do with my changes. I tried building `main` and running it on device and it is also happening there.

File: src/pages/home/report/ReportFooter.js
Comment (ospfranco) [reply src/pages/home/report/ReportFooter.js]: done :)

**Comment by mvtglobally** [comment]
QA is completed

File: src/pages/home/ReportScreen.js
Comment (ospfranco) [reply src/pages/home/ReportScreen.js:171]: The problem with that first render is that it would delay the rendering one extra cycle at which point onyx would be already trying to shove data into the component and would delay the rendering again. However I don't think there is a way around it at this point. I will just revert it and we can revisit this again later.

**Comment by ospfranco** [comment]
@fedirjh I've reverted the change to the `isLoading` flag. I've tested it with my account but I'm not sure if I can reproduce it with any of the chats I have. Can you take a look and confirm it is working?

**Comment by ospfranco** [comment]
damnit haha let me think about it for a bit, if you have any ideas it would be greatly appreciated as well

File: src/pages/home/ReportScreen.js
Comment (ospfranco) [reply src/pages/home/ReportScreen.js]: ah, you mean a different flag?

```
const isLoading = !reportID || !isSidebarLoaded || _.isEmpty(personalDetails);
const shouldShowNotFoundPage = !reportID || !isSidebarLoaded || _.isEmpty(personalDetails) || firstRenderRef.current;
```

**Review by fedirjh — APPROVED** [review]
All yours @mountiny.

File: src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js
Comment (mountiny) [inline src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js:56]: Does not seem to have a default value set, also there are required params coming after
```suggestion
 * @param {Boolean} ignoreDefaultRooms
```

File: src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js
Comment (mountiny) [inline src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js:100]: This is debug no?

File: src/pages/home/report/ReportActionCompose/SilentCommentUpdater.js
Comment (mountiny) [reply src/pages/home/report/ReportActionCompose/SilentCommentUpdater.js:71]: Did you forget to apply this suggestion?

File: src/pages/home/report/ReportActionsList.js
Comment (mountiny) [inline src/pages/home/report/ReportActionsList.js:181]: I think we should have an early return here as that is preferred

File: src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js
Comment (mountiny) [inline src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js:77]: This is a new component and we prefer destructuring the props, can you please do that here?

**Review by mountiny — CHANGES_REQUESTED** [review]

File: src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js
Comment (ospfranco) [reply src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js:56]: I'm not sure I understand. This is just an internal function and doesn't require a default value, it should always be called with a boolean value:

CleanShot 2023-09-27 at 08 15 37

File: src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js
Comment (ospfranco) [reply src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js:100]: Yeah, can be removed

File: src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js
Comment (mountiny) [reply src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js]: Yeah but the `[]` in js docs means optional parameter, but this is not optional parameter but the looks of things

Comment (ospfranco) [reply src/libs/Navigation/AppNavigator/ReportScreenIDSetter.js]: Ah, this wasn't my code, just moved it from the wrapper. But it is fixed.

**Review by mountiny — APPROVED** [review]
Thanks! looking great!

**Comment by ospfranco** [comment]
Ah, unit test is failing, let me take a look

File: src/ONYXKEYS.ts
Comment (roryabraham) [inline src/ONYXKEYS.ts:245]: Request: can we please add a comment to explain what belongs in `report_` and what belongs in `reportMetadata_`. Without all the context the distinction isn't very clear.

Comment (ospfranco) [reply src/ONYXKEYS.ts:245]: For now we've separated the loading states `isLoadingReportActions` and `isLoadingMoreReportActions` into `reportMetadata_`, as they was causing many re-renders on smaller components every time the report actions were being fetched. It makes sense to add it a comment for future reference.

Comment (roryabraham) [reply src/ONYXKEYS.ts:245]: Thanks, sounds good. Maybe the distinction is that REPORT_METADATA is for client-generated data?

Comment (ospfranco) [reply src/ONYXKEYS.ts:245]: It's a performance optimization, not a hard cut between functionality/logic.

**Pull request merged**
